marshal.go

```diff
@@ -280,3 +286,15 @@ var formSetters = []func(string, string, *Params){
 	},
 	sourceBody: nil,
 }
+
+// BytesReaderCloser is similar to the type returned by
```

Comment:
I don't like this comment it reads more like a justification / moan than a description. how about:
```
// BytesReaderCloser is a bytes.Reader which 
// implements io.Closer with a no-op Close method.
```